cadc-dali-parquet/src/main/java/ca/nrc/cadc/dali/tables/parquet/io/HttpRandomAccessSource.java

```diff
@@ -0,0 +1,67 @@
+package ca.nrc.cadc.dali.tables.parquet.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class HttpRandomAccessSource implements RandomAccessSource {
+    private final URL url;
+    private long position = 0;
+    private final long contentLength;
+
+    public HttpRandomAccessSource(URL url) throws IOException {
+        this.url = url;
+        this.contentLength = fetchContentLength(url);
+    }
+
+    @Override
+    public void seek(long position) throws IOException {
+        if (position < 0 || position >= contentLength) {
+            throw new IOException("Seek position out of bounds");
+        }
+        this.position = position;
+    }
+
+    @Override
+    public int read(byte[] buffer, int offset, int length) throws IOException {
+        long end = position + length - 1;
+
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        conn.setRequestProperty("Range", "bytes=" + position + "-" + end);
+        conn.connect();
+
+        try (InputStream in = conn.getInputStream()) {
+            int bytesRead = in.read(buffer, offset, length);
+            if (bytesRead > 0) {
+                position += bytesRead;
+            }
+            return bytesRead;
+        } finally {
+            conn.disconnect();
+        }
+    }
+
+    @Override
+    public long length() throws IOException {
+        return contentLength;
+    }
+
+    @Override
+    public void close() {
+        // No persistent connection to close
+    }
+
+    private long fetchContentLength(URL url) throws IOException {
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
ca.nrc.cadc.net.HttpGet can perform head requests

also, this head request should check the http headers to see if the server accepts range requests:
```
accept-ranges: bytes
```
I can't recall what happens in a GET request with range header if the server does not support ranges.

aside: I see that our `minoc` services to not include `accept-ranges` even though they do; I'll create a bug report for that as it will matter.